sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/tests/PlaywrightReporterTests.cs

```diff
@@ -0,0 +1,266 @@
+            environmentMock.Verify(e => e.Exit(1), Times.Once);
+            Assert.Null(reporter._testProcessor);
+        }
+
+        [Test]
+        public void InitializePlaywrightReporter_WithNoServiceUrl_PrintsMessageAndExits()
+        {
+            var consoleWriterMock = new Mock<IConsoleWriter>();
+            var environmentMock = new Mock<IEnvironment>();
+
+            environmentMock.Setup(e => e.Exit(It.IsAny<int>())).Callback<int>(i => { });
+            consoleWriterMock.Setup(c => c.WriteError(It.IsAny<string>())).Verifiable();
+
+            var reporter = new PlaywrightReporter(logger: null, environment: environmentMock.Object, xmlRunSettings: null, consoleWriter: consoleWriterMock.Object, jsonWebTokenHandler: null);
+
+            reporter.InitializePlaywrightReporter("");
+
+            consoleWriterMock.Verify(c => c.WriteError(Constants.s_no_service_endpoint_error_message), Times.Once);
+            environmentMock.Verify(e => e.Exit(1), Times.Once);
+            Assert.Null(reporter._testProcessor);
+        }
+
+        [Test]
+        public void InitializePlaywrightReporter_WithNoAccessToken_PrintsMessageAndExits()
+        {
+            var consoleWriterMock = new Mock<IConsoleWriter>();
+            var environmentMock = new Mock<IEnvironment>();
+
+            environmentMock.Setup(e => e.Exit(It.IsAny<int>())).Callback<int>(i => { });
+            environmentMock.Setup(e => e.GetEnvironmentVariable(ReporterConstants.s_pLAYWRIGHT_SERVICE_REPORTING_URL)).Returns("reporting-url");
+            consoleWriterMock.Setup(c => c.WriteError(It.IsAny<string>())).Verifiable();
+
+            var reporter = new PlaywrightReporter(logger: null, environment: environmentMock.Object, xmlRunSettings: null, consoleWriter: consoleWriterMock.Object, jsonWebTokenHandler: null);
+
+            reporter.InitializePlaywrightReporter("");
+
+            consoleWriterMock.Verify(c => c.WriteError(Constants.s_no_auth_error), Times.Once);
+            environmentMock.Verify(e => e.Exit(1), Times.Once);
+            Assert.Null(reporter._testProcessor);
+        }
+
+        [Test]
+        public void InitializePlaywrightReporter_Default_SetsUpTestProcessor()
+        {
+            var consoleWriterMock = new Mock<IConsoleWriter>();
+            var environmentMock = new Mock<IEnvironment>();
+            var accessToken = GetToken(new Dictionary<string, object> { { "aid", "account-id-guid" }, { "oid", "org-id" }, { "id", "uuid" }, { "name", "username" } });
+
+            environmentMock.Setup(e => e.GetEnvironmentVariable(ServiceEnvironmentVariable.PlaywrightServiceRunId)).Returns("run-id");
+            environmentMock.Setup(e => e.GetEnvironmentVariable(ReporterConstants.s_pLAYWRIGHT_SERVICE_REPORTING_URL)).Returns("https://eastus.reporting.api.playwright.microsoft.com");
+            environmentMock.Setup(e => e.GetEnvironmentVariable(ServiceEnvironmentVariable.PlaywrightServiceAccessToken)).Returns(accessToken);
```

Comment:
i assume this Entra Token in default flow, should we make it clear ?

Reply:
Those tests are in PlaywrightServiceTests file